### ISSUE TITLE
Default arguments for flatten overfit in TESS light curves

### ISSUE THREAD

**Issue by mirca** [KeplerGO/lightkurve#546]
#### Problem description
While removing long term trends of TESS light curves, I've noticed that the current default arguments for ``flatten`` usually overfit transits.

#### Example

```python
import lightkurve as lk

pixels = pixels = lk.search_targetpixelfile("wasp-124b").download()
lightcurve = pixels.to_lightcurve()
flat_lightcurve, trend  = lightcurve.flatten(return_trend = True)
trend.plot()
exoplanet = flat_lightcurve.fold(period=3.372)
exoplanet.plot()
```
![Screen Shot 2019-07-12 at 8 31 02 PM](https://user-images.githubusercontent.com/13077051/61163400-15fe5900-a4e4-11e9-8095-63579958e4d0.png)

#### Expected behavior

Increasing the break_tolerance and the window length helps a lot:
```python
flat_lightcurve, trend  = lightcurve.flatten(window_length = 1001, break_tolerance = 50, return_trend = True)
trend.plot()
exoplanet = flat_lightcurve.fold(period=3.372)
exoplanet.plot()
```
![Screen Shot 2019-07-12 at 8 31 20 PM](https://user-images.githubusercontent.com/13077051/61163426-49d97e80-a4e4-11e9-8e46-309c9da735a9.png)
I couldn't verify this overfit behavior in Kepler/K2 lightcurves.

#### Environment:
- platform (e.g. Linux, OSX, Windows): OSX
- lightkurve version (e.g. 1.0b6): I'm living in the bleeding edge (1.1.dev) :D
- installation method (e.g. pip, conda, source): source

**Comment by barentsen** [comment]
Right!!  I guess we should really either change the defaults for TESS, or parameterize it in units of time instead of cadences!  (Because the problem here is that the data has 2 minute cadence, I think!)

**Comment by mirca** [comment]
@barentsen it seems that ``break_tolerance = 50`` doesn't break anything for Kepler/K2 light curves.  I'm investigating :)

**Comment by hippke** [comment]
I agree: The ``window_length`` should be in units of time (also ``break_tolerance``). Inspired by your API, I used the same keywords in my [wotan](https://github.com/hippke/wotan) package. Hope that's ok :-)

I also recommend replacing the savgol filter, as it performed amongst the worst of all I tested. If you want a good but simple solution, I recommend a robustified "sum of sines" filter, which also cooperates nicely with a ``transit_mask`` to mask off known transits (see example [here](https://github.com/hippke/wotan/blob/master/examples/transit%20mask.ipynb)). 

The robustified sum-of-sines (with iterative sigma-clipping) and the transit mask filter are pure Python with only ~30 SLOC. If you're interested I think that should be straightforward to implement in ``lightkurve``.

**Comment by barentsen** [comment]
@hippke We should definitely incorporate the results of your study in Lightkurve!!  I wonder what the best approach is, e.g.:
1. We could add `wotan` as a dependency and enable users to use any wotan-implemented method, though we probably wouldn't be able to tune or test them all in detail.
2. Alternatively, we could pick 2 or 3 key methods from `wotan` paper, move them into Lightkurve, and test+support them very carefully.  (Which 2 or 3 methods would be best?)

Do you have thoughts?

**Comment by hippke** [comment]
I think the best approach for ``lightkurve`` would be to integrate (copy-paste) small parts of the ``wotan`` code, so that another dependency can be avoided.

As follows I recommend some methods and explain a few details:

1. Robust slider using the biweight
   - Winner of blind transit search
   - Pure python <50 SLOC, but slow. 100x faster with ``numba`` or ``Cython``. Is that an option?
   - Doesn't work for masked transits if the transit duration (=mask) is longer than ``window_length``
2. Sum of sines using iterative sigma-clipping
   - It's a global method (all data is considered at once), thus complementary to a slider
   - Many cases of stellar variability (including strong variability, e.g. pulsations) are well modelled with sines
   - Core functionality is in ``np.matmul`` and ``np.linalg.lstsq``, but building the matrix fast requires ``numba`` or ``Cython``
   - *Does* work reasonably well for masked transits if the transit duration (=mask) is longer than ``window_length``. That's because all data is considered at once, and a bunch of gaps is not that critical
3. A robust spline
   - ``wotan`` has a robust (iterative sigma-clipping) implementation based on ``scipy``. No need for ``numba`` or ``Cython``
   - The huber-spline (more elegant because one-pass) gives similar results, but depends on ``scikit-learn``
   - Global approach, sharing the "sum of sines" advantages

After these three, it gets more exotic and specialized.

What I would *not* recommend for ``lightkurve``:
- GPs. Almost every feedback or question I get mentions GPs. For a blind transit search, I found that they are not a good choice. For other purposes, I have an opinion "which this margin is too narrow to contain." Although ``wotan`` has a few GP options (from ``scikit-learn``), and even a periodogram-driven implementation, I would point heavy users rather towards a specialized package.
- A penalized spline for automatic knot distance determination (i.e. ``window_length``). Tempting to use, as it avoids having to choose the window size. That's relevant even if we parametrize in units of time: different data need different windows. In practice, however, the automatic determination is often not very good. That could lead to much user dissatisfaction. There's also a dependency to ``pygam``.

**Comment by gully** [comment]
Note this is near-duplicate of #486

**Comment by orionlee** [comment]
On a related note, some other methods' default parameters might be tailored to Kepler as well.

1. `transit_duration` for `LightCurve.estimate_cdpp()` 
https://docs.lightkurve.org/api/lightkurve.lightcurve.LightCurve.html#lightkurve.lightcurve.LightCurve.estimate_cdpp
